### ISSUE TITLE
[v6][database] snapshot values not in the correct order

### ISSUE THREAD

**Issue by losheredos** [invertase/react-native-firebase#2448]
## Issue

I have started to use RN version 0.60.4 , I was using 0.57.5 before. I use firebase real time database. I was fetching last 50 data, now I see that it brings me last 50 but in random order.. I dont understand the reason of it. I had no problem in installlation, all works fine but just fetching data brings random last 50..

Platform: Android

I have checked if its from Flatlist or something but its not from those.. In old version its working fine, but now in latest RN version it works wrong.

package.json

```
"dependencies": {
    "@react-native-community/async-storage": "^1.6.1",
    "@react-native-firebase/app": "^0.1.2",
    "@react-native-firebase/database": "^0.1.2",
    "axios": "^0.19.0",
    "immutable": "^4.0.0-rc.12",
    "react": "16.8.6",
    "react-native": "0.60.4",
    "react-native-device-info": "^2.3.2",
    "react-native-gesture-handler": "^1.3.0",
    "react-native-iap": "^3.3.7",
    "react-native-image-picker": "^1.0.2",
    "react-native-indicators": "^0.13.0",
    "react-native-onesignal": "^3.3.1",
    "react-native-progress": "^3.6.0",
    "react-native-router-flux": "^4.0.6",
    "react-native-sound": "^0.11.0",
    "react-navigation": "^3.11.1",
    "react-redux": "^7.1.0",
    "redux": "^4.0.4",
    "redux-logger": "^3.0.6"
  },

```
My Code:

```

import { firebase } from '@react-native-firebase/database';

// CODE //
 firebase.database().ref('Chat/'+LL).limitToLast(50).on('value', function (snapshot) {
            let array = [];
            let data = snapshot.val()

            Object.keys(data).map(function(key, index) {
                console.log(data[key])
                array.push(data[key])
            });
            this.setState({ chats: array, loaded: true });
        }.bind(this));
```

**Comment by Salakar** [comment]
Hey @losheredos,

Can you try with `snapshot.forEach()` -> this guarantees the order, whereas `snapshot.val()` keys won't - see the [docs here for `forEach()`](https://invertase.io/oss/react-native-firebase/v6/database/reference/datasnapshot#forEach)

So your code above becomes: 

```js
firebase
  .database()
  .ref('Chat/' + LL)
  .limitToLast(50)
  .on('value', snapshot => {
    let array = [];
    snapshot.forEach(function(childSnapshot) {
      // const key = childSnapshot.key;
      const childData = childSnapshot.val();
      console.log(childData);
      array.push(childData);
    });
    this.setState({ chats: array, loaded: true });
  });
```

**Comment by losheredos** [comment]
Hey @Salakar ,

I actually tried it just forgot to mention it. When I try your suggestion which I also tried before, I get error: 

this._ref.child is not a function.

**Comment by Ehesp** [comment]
Hey, could you send over a sample data set and I'll get this sorted tomorrow.

**Comment by losheredos** [comment]
Fetching: 
![resim](https://user-images.githubusercontent.com/24756127/62580515-f5d97400-b8ae-11e9-8058-e5e0bfd5409a.png)

Flatlist:
![resim](https://user-images.githubusercontent.com/24756127/62580528-012c9f80-b8af-11e9-9ac7-4f25af41fe0e.png)

RenderItem:
![resim](https://user-images.githubusercontent.com/24756127/62580549-0db0f800-b8af-11e9-80a2-a5339be31cb5.png)

**Comment by Ehesp** [comment]
Thanks, will get this looked at

**Comment by Salakar** [comment]
Which platform is this as well, Android or iOS? Thanks

**Comment by losheredos** [comment]
Sorry that didnt mention it, its Android

**Comment by Ehesp** [comment]
@losheredos what does the data in the database look like? Do you have a list of generated keys with text or is it more complex? I ran a basic test and it works my end, so assume it's the data being in a different format.

**Comment by losheredos** [comment]
`Console.log(snapshot.val())`
![resim](https://user-images.githubusercontent.com/24756127/62610347-30bdc500-b90c-11e9-945f-c3e50a2da624.png)

**Comment by Ehesp** [comment]
Ok makes more sense, can you give me an example of what one of them objects looks like? Replace any sensitive data with something else of course :D

**Comment by losheredos** [comment]
I have edited. Look the case is in my current app which is online now, chat works fine with as in correct order. But now in when I work with RN 0.60.4, they are messed up and mixed. The main difference is I cant forEach snapshot directly without getting val() of it.

**Comment by Ehesp** [comment]
Ok sorted it - even has a comment on the code with the issue... woops!

So anyway to answer your questions: 

- When doing `Object.keys` or `Object.values`, you're iterating over the data we get back from the native calls. When this comes to JavaScript, the order can be anything - hence why it wont work.

- When doing `forEach` (which is now fixed), we iterate over an array of the keys which is sent from native which is in the correct order, and grab the data from the object it relates to.

Will get a version pushed up soon (couple of other things to sort too). Thanks for your help.

**Comment by Salakar** [comment]
This has now been published in 0.1.3 of the new packages, see https://github.com/invertase/react-native-firebase/issues/2025#issuecomment-519166641.

Thanks
